src/cdk-experimental/testing/component-harness.ts

```diff
@@ -0,0 +1,220 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/**
+ * Test Element interface
+ * This is a wrapper of native element
+ */
+export interface TestElement {
+  blur(): Promise<void>;
+  clear(): Promise<void>;
+  click(): Promise<void>;
+  focus(): Promise<void>;
+  getCssValue(property: string): Promise<string>;
+  hover(): Promise<void>;
+  sendKeys(keys: string): Promise<void>;
+  text(): Promise<string>;
+  getAttribute(name: string): Promise<string|null>;
+}
+
+/**
+ * Extra searching options used by searching functions
+ *
+ * @param allowNull Optional, whether the found element can be null. If
+ * allowNull is set, the searching function will always try to fetch the element
+ * at once. When the element cannot be found, the searching function should
+ * return null if allowNull is set to true, throw an error if allowNull is set
+ * to false. If allowNull is not set, the framework will choose the behaviors
+ * that make more sense for each test type (e.g. for unit test, the framework
+ * will make sure the element is not null; otherwise throw an error); however,
+ * the internal behavior is not guaranteed and user should not rely on it. Note
+ * that in most cases, you don't need to care about whether an element is
+ * present when loading the element and don't need to set this parameter unless
+ * you do want to check whether the element is present when calling the
+ * searching function. e.g. you want to make sure some element is not there when
+ * loading the element in order to check whether a "ngif" works well.
+ *
+ * @param global Optional. If global is set to true, the selector will match any
+ *     element on the page and is not limited to the root of the harness. If
+ * global is unset or set to false, the selector will only find elements under
+ * the current root.
+ */
+export interface Options {
+  allowNull?: boolean;
+  global?: boolean;
+}
+
+/**
+ * Type narrowing of Options to allow the overloads of ComponentHarness.find to
+ * return null only if allowNull is set to true.
+ */
+interface OptionsWithAllowNullSet extends Options {
+  allowNull: true;
+}
+
+/**
+ * Locator interface
+ */
+export interface Locator {
```

Comment:
I moved `TestElement` to its own file, but I think its more trouble than its worth for `Locator` since the `Locator` and `ComponentHarness` types depend on each other